mentoring/dashboard.py

```diff
@@ -0,0 +1,385 @@
+                        _("Invalid color rule on line {line_number}").format(line_number=lineno + 1)
+                    )
+        return rules
+
+    @lazy
+    def color_rules_parsed(self):
+        """
+        Caching property to get parsed color rules. Returns a list of ColorRule objects.
+        """
+        return self.parse_color_rules_str(self.color_rules) if self.color_rules else []
+
+    def _get_submission_key(self, usage_key):
+        """
+        Given the usage_key of an MCQ block, get the dict key needed to look it up with the
+        submissions API.
+        """
+        return dict(
+            student_id=self.runtime.anonymous_student_id,
+            course_id=unicode(usage_key.course_key),
+            item_id=unicode(usage_key),
+            item_type=usage_key.block_type,
+        )
+
+    def color_for_value(self, value):
+        """ Given a string value, get the color rule that matches, if any """
+        if isinstance(value, basestring):
+            if value.isnumeric():
+                value = float(value)
+            else:
+                return None
+        for rule in self.color_rules_parsed:
+            if rule.matches(value):
+                return rule.color_str
+        return None
+
+    def generate_content(self, include_report_link=True):
+        """
+        Create the HTML for this block, by getting the data and inserting it into a template.
+        """
+        blocks = []
+        for mentoring_block in self.get_mentoring_blocks(self.mentoring_ids):
+            if mentoring_block is None:
+                continue
+            block = {
+                'display_name': mentoring_block.display_name,
+                'mcqs': []
+            }
+            for child_id in mentoring_block.children:
+                if child_isinstance(mentoring_block, child_id, MCQBlock):
+                    # Get the student's submitted answer to this MCQ from the submissions API:
+                    mcq_block = self.runtime.get_block(child_id)
+                    mcq_submission_key = self._get_submission_key(child_id)
+                    try:
+                        value = sub_api.get_submissions(mcq_submission_key, limit=1)[0]["answer"]
+                    except IndexError:
+                        value = None
+                    block['mcqs'].append({
+                        "display_name": mcq_block.question,
+                        "value": value,
+                        "color": self.color_for_value(value),
+                    })
+            # If the values are numeric, display an average:
+            numeric_values = [float(mcq['value']) for mcq in block['mcqs'] if mcq['value'] and mcq['value'].isnumeric()]
+            if numeric_values:
+                average_value = sum(numeric_values) / len(numeric_values)
+                block['average'] = average_value
+                block['has_average'] = True
+                block['average_color'] = self.color_for_value(average_value)
```

Comment:
What references 'average', 'has_average', and 'average_color'? I don't see any code actually inspecting these atrributes